### PR TITLE
CSP-95-Outer-TrainingLocationsForCourse

### PR DESCRIPTION
**Pull request by raviudari01**

### CODE REVIEW

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProviderCourseLocation/GetProviderCourseLocationQueryHandler.cs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            if (providerCourseLocationsResponse.StatusCode != HttpStatusCode.OK)
+            {
+                var errorMessage =
+                   $"Response status code does not indicate success: {(int)providerCourseLocationsResponse.StatusCode} - Provider Course locations not found for ukprn: {request.Ukprn} LarsCode: {request.LarsCode}";
+                _logger.LogError(errorMessage);
+                throw new HttpRequestContentException(errorMessage, providerCourseLocationsResponse.StatusCode, providerCourseLocationsResponse.ErrorContent);
+            }
+            var providerCourseLocations = providerCourseLocationsResponse.Body;
+            
+            var locations = providerCourseLocations.Where(a => a.LocationType == LocationType.Provider).Select(x => (ProviderCourseLocationModel)x).ToList();
```
Comment (DevenKShah) [inline src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProviderCourseLocation/GetProviderCourseLocationQueryHandler.cs:42]: There is no coverage for this linq statement.

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api.UnitTests/Controllers/ProviderCourseLocationControllerTests.cs [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+
+            var response = await sut.GetProviderCourseLocations(ukprn, larsCode);
+
+            var okResult = response as OkObjectResult;
+            okResult.Should().NotBeNull();
+            okResult.Value.Should().BeEquivalentTo(result);
+        }
+
+        [Test, MoqAutoData]
+        public async Task GetProviderCourseLocations_InvalidRequest_ReturnsLocations(
```
Comment (DevenKShah) [inline src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api.UnitTests/Controllers/ProviderCourseLocationControllerTests.cs:38]: Test name is wrong, it returns not found

**Review by DevenKShah — APPROVED** [review]
Some coverage is missing but looks good otherwise.

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCourseLocationController.cs [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.Extensions.Logging;
+using SFA.DAS.RoatpCourseManagement.Application.Standards.Queries.GetProviderCourseLocation;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace SFA.DAS.RoatpCourseManagement.Api.Controllers
+{
+    [ApiController]
+    public class ProviderCourseLocationController : ControllerBase
```
Comment (DevenKShah) [inline src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCourseLocationController.cs:11]: ProviderCourseLocationsController

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCourseLocationController.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        private readonly ILogger<ProviderCourseLocationController> _logger;
+
+        public ProviderCourseLocationController(ILogger<ProviderCourseLocationController> logger, IMediator mediator)
+        {
+            _logger = logger;
+            _mediator = mediator;
+        }
+
+        [HttpGet]
+        [Route("providers/{ukprn}/courses/{larsCode}/provider-course-locations")]
```
Comment (DevenKShah) [inline src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCourseLocationController.cs:23]: There is an inconsistency between this and work done in CSP-114. You have a controller `ProviderCourseLocationRegionsController` in there and the endpoint has `standardsubregions` on it. 
I think we should have only one Controller `ProviderCourseLocationsController` and this should have only one endpoint that looks like this... `providers/{ukprn}/courses/{larsCode}/locations?{locationtype}`.

**Review by DevenKShah — CHANGES_REQUESTED** [review]

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProviderCourseLocation/GetProviderCourseLocationQueryHandler.cs (hunk 1 above)
Comment (DevenKShah) [reply src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProviderCourseLocation/GetProviderCourseLocationQueryHandler.cs:42]: Can you point me to the test where you have got different type of locations and only provider type are being returned please, as I am not able to find one.

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCourseLocationController.cs (hunk 4 above)
Comment (DevenKShah) [reply src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement.Api/Controllers/ProviderCourseLocationController.cs]: ok I see, the `standardssubregions` is doing a little bit more than just getting locations for the course. However the endpoint names are inconsistent. This one needs to be updated to `providers/{ukprn}/courses/{larsCode}/locations/provider-locations`.

File: src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProviderCourseLocation/GetProviderCourseLocationQueryHandler.cs (hunk 1 above)
Comment (MarkFCain) [reply src/RoatpCourseManagement/SFA.DAS.RoatpCourseManagement/Application/Standards/Queries/GetProviderCourseLocation/GetProviderCourseLocationQueryHandler.cs:42]: @raviudari01 ping me, I'll can see the shortfall

**Review by MarkFCain — APPROVED** [review]

**Review by DevenKShah — APPROVED** [review]

**Pull request merged**